src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4517,6 +4517,11 @@ private ParameterSyntax ParseParameter()
                         this.GetExpectedTokenError(expected: SyntaxKind.ExclamationExclamationToken, actual: firstExclamation.Kind));
                 }
             }
+            else if (this.CurrentToken.Kind is SyntaxKind.ExclamationExclamationToken)
+            {
+                // This is reachable for incremental parsing.
+                exclamationExclamationToken = EatToken();
```

Comment:
I have mixed feelings. Not in front of a computer, but I feel like we have some logic elsewhere (maybe in the blender) for dealing with merged tokens (like `>>`). I think that same location should be updated for this special token.

Reply:
Actually, this is certainly wrong.  because it means that if the edit to the tree is such that the synthesized `!!` gets moved somewhere else (e.g. to now be a suffix suppression operator), that code will break because it won't see this as two exclamation tokens, but a single token as well.  These merged tokens can't be returned from the incremental lexer as they do not correspond to what the lexer would have produced originally.